Source/ACE/Entity/Player.cs

```diff
@@ -1644,9 +1672,78 @@ public void HandleActionPutItemInContainer(ObjectGuid itemGuid, ObjectGuid conta
             putItemInContainerChain.EnqueueChain();
         }
 
+        private void UpdateInventory(bool intoWorld, WorldObject inventoryItem)
+        {
+            if (intoWorld)
+            {
+                inventoryItem.Location = PhysicsData.Position.InFrontOf(1.1f);
+                inventoryItem.PositionFlag = UpdatePositionFlag.Contact
+                                             | UpdatePositionFlag.Placement
+                                             | UpdatePositionFlag.ZeroQy
+                                             | UpdatePositionFlag.ZeroQx;
+
+                inventoryItem.PhysicsData.PhysicsDescriptionFlag = inventoryItem.PhysicsData.SetPhysicsDescriptionFlag(inventoryItem);
+                inventoryItem.ContainerId = null;
+                inventoryItem.Wielder = null;
+                inventoryItem.WeenieFlags = inventoryItem.SetWeenieHeaderFlag();
+            }
+            else
+            {
+                inventoryItem.ContainerId = Guid.Full;
+                if (inventoryItem.Location != null)
+                    LandblockManager.RemoveObject(inventoryItem);
+                inventoryItem.PhysicsData.PhysicsDescriptionFlag &= ~PhysicsDescriptionFlag.Position;
+                inventoryItem.PositionFlag = UpdatePositionFlag.None;
+                inventoryItem.PhysicsData.Position = null;
+                inventoryItem.Location = null;
+                inventoryItem.WeenieFlags = inventoryItem.SetWeenieHeaderFlag();
+            }
+        }
+
+        public void HandleActionPutItemInContainer(ObjectGuid itemGuid, ObjectGuid containerGuid, uint location = 0)
```

Comment:
This function looks at the player.  To do so safely it needs to run its operations as part of the player's actions.  (e.g. using ActionChains).  Its also prefixed with HandleAction indicating its safe to call from outside the player's action context (so it must use action chains).